### DISCUSSION TITLE
Create your own effect

### DISCUSSION THREAD

**Discussion by thelvyn777** [awawa-dev/HyperHDR#59] [Q&A]
Hello,

I want to create/update/delete effect.
Where are the files to update  stored  ?

**Comment by awawa-dev** [comment]
Hi
If your are looking for an old python effect module, it was abandon in favor of build-in effects.

**Reply by thelvyn777** [reply]
I thought , it could be done after reading this : "https://docs.hyperion-project.org/en/effects/OurFirstEffect.htm" but didnt find the folder.

What's the method now to create or update an effect ?

**Reply by awawa-dev** [reply]
Effect's module in HyperHDR was completely rewritten to save resources and to provide better stability on different systems: the way it was integrated before with Python led to different problem and unpredictable behavior.  Although effects has a high priority for this project  (for ex. new music effects) better performance and stable behavior for usage with USB grabber for movie content has a higher priority so no going back. Beside low latency routing was a mission impossible for the old way and music effects require that.

And to answer your second question it's still possible to create new effects but it need to be done in the code (c++) and compile with the project. You could take a look for example on https://github.com/awawa-dev/HyperHDR/blob/master/libsrc/effectengine/Animation_Police.cpp to see how it works and how it was upgraded from old Python script and to see if you feel up to it. I can provide help for particular cases with creating new interesting effects (how to anchor them in the API) but it can be limited by my spare time ;-)

**Reply by thelvyn777** [reply]
Thank You
